tests/test.py

```diff
@@ -0,0 +1,192 @@
+    # "macro_operator_precedence.c": "macro_operator_precedence.csv",
+    # "macro_test.c": "macro_test.csv",
+    "omitted_curly_braces.c": "omitted_curly_braces.csv",
+    # "operator_precedence.c": "operator_precedence.csv",
+    "post_incdec.c": "post_incdec.csv",
+    "pre_incdec.c": "pre_incdec.csv",
+    "repurposed_variable.c": "repurposed_variable.csv",
+    "reversed_subscripts.c": "reversed_subscripts.csv",
+    "type_conversion.c": "type_conversion.csv",
+}
+
+patch_path = Path.cwd() / "cocci"
+input_path = Path.cwd() / "tests/data/inputs"
+expected_path = Path.cwd() / "tests/data/expected_outputs"
+
+def compare(patch, input, verbosity):
+    try:
+        data = run_cocci(["--quiet"], patch_path / patch, input_path / input)
+
+        expected = expected_path / input_to_expected_mapping[input]
+        
+        output = read_csv(data, header=None) # turns StringIO result into dataframe
+        
+        expected = read_csv(expected, header=None)
+        
+        expected = expected.iloc[:,[2,4]]
+        output = output.iloc[:,[2,4]]
+
+        for i in range(len(output)):
+            output.iloc[i,1] = output.iloc[i,1].replace(' ', '')
+
+        output = output.groupby(2)[4].apply(list).reset_index()
+        expected = expected.groupby(2)[4].apply(list).reset_index()
+        combined = pd.merge(expected, output, on=2, suffixes=('_1', '_2'), how="outer")
+
+        # pd.set_option('display.max_rows', None)
+        # pd.set_option('display.max_columns', None)
+
+        combined = combined.fillna(pd.NA) # otherwise the next line won't work
+        combined = combined.map(lambda x: [] if x is pd.NA else x)
+
+        combined.to_csv("tests/debug.csv", index=False)
+        # strings1 should be expected, and s2 is output
+        def is_contained(strings1, strings2):
+            set1 = set(strings1)
+            set2 = set(strings2)
+            return set1 == set2
+            # return (len(set1)==len(set2))&all(any(s1 in s2 for s1 in set1) for s2 in set2)
+        
+        combined['Containment'] = combined.apply(lambda row: is_contained(row['4_1'], row['4_2']), axis=1)        
+        is_equal = combined['Containment'].all()
+        return is_equal
+    
+    except Exception as e:
+        print(f"Something else happened: {e}, for {input}")
+
+# individual tests
+@pytest.mark.parametrize("input", patch_to_input_mapping["assignment_as_value.cocci"])
+def test_assignment_as_value(input):
+    patch = "assignment_as_value.cocci"
+    # missing_in_expected, missing_in_output = compare(patch, input, 0)
+    result = compare(patch, input, 0)
+    assert(result == True)
+    # assert(not len(missing_in_output) and not len(missing_in_expected) == True)
+
+@pytest.mark.parametrize("input", patch_to_input_mapping["change_of_literal_encoding.cocci"])
+def test_change_of_literal_encoding(input):
+    patch = "change_of_literal_encoding.cocci"
+    result = compare(patch, input, 0)
+    assert(result == True)
```

Comment:
`assert` could be directly inside compare. `is_equal = combined['Containment'].all()` `assert is_equal`